internal/venice-common/src/main/java/com/linkedin/venice/helix/ZkRoutersClusterManager.java

```diff
@@ -281,10 +293,13 @@ protected void triggerRouterClusterConfigChangedEvent(RoutersClusterConfig newCo
    * will be executed to process the event.
    */
   @Override
-  public void handleChildChange(String parentPath, List<String> currentChilds) throws Exception {
+  public void handleChildChange(String parentPath, List<String> currentChildren) throws Exception {
     int oldLiveRouterCount = liveRouterCount;
-    changeLiveRouterCount(currentChilds.size());
-    LOGGER.info("Live router count has been changed from: {} to: {}.", oldLiveRouterCount, currentChilds.size());
+    liveRouterInstanceSet.clear();
+    liveRouterInstanceSet
```

Comment:
`clear` and `addAll` can be risky when some user is trying to read all the live Router instances. 
Why not just swap it instead?
Essentially, we can create a new instance set and assign it to `livceRouterInstanceSet` once it is completed.